app/Actions/Diagnostics/Pipes/Checks/AppUrlMatchCheck.php

```diff
@@ -20,67 +24,66 @@ class AppUrlMatchCheck implements DiagnosticPipe
 	public function handle(array &$data, \Closure $next): array
 	{
 		$config_url = config('app.url');
+		$dir_url = config('app.dir_url');
 		if (config('app.url') === 'http://localhost') {
-			$data[] = 'Warning: APP_URL is still set to default, this will break access to all your images and assets if you are using Lychee behind a sub-domain.';
+			$data[] = 'Warning: APP_URL is still set to default, this will break access to all your images';
+			$data[] = self::INVISIBLE_WARNING . 'and assets if you are using Lychee behind a sub-domain.';
 		}
 
 		$bad = $this->splitUrl($config_url)[3];
 
-		$censored_bad = $this->censor($bad);
+		$censored_bad = Helpers::censor($bad);
 		$censored_app_url = $this->getCensorAppUrl();
 		$censored_current = $this->getCensorCurrentUrl();
 
 		if ($bad !== '') {
 			$data[] = sprintf(
-				'Warning: APP_URL (%s) contains a sub-path (%s). This may impact your WebAuthn authentication.',
+				'Error: APP_URL (%s) contains a sub-path (%s).',
+				$censored_app_url,
+				$censored_bad,
+			);
+			$data[] = sprintf(
+				self::INVISIBLE_ERROR . 'Instead set APP_DIR to (%s) and APP_URL to (%s) in your .env',
+				$censored_bad,
+				$censored_current,
+			);
+		}
+
+		if ($bad !== '') {
+			$data[] = sprintf(
+				'Warning: APP_URL (%s) contains a sub-path (%s).',
 				$censored_app_url,
-				$censored_bad);
+				$censored_bad
+			);
+			$data[] = self::INVISIBLE_WARNING . 'This may impact your WebAuthn authentication.';
 		}
 
 		if (!$this->checkUrlMatchCurrentHost()) {
 			$data[] = sprintf(
-				'Error: APP_URL (%s) does not match the current url (%s). This will break WebAuthn authentication.',
+				'Error: APP_URL (%s) does not match the current url (%s).',
 				$censored_app_url,
 				$censored_current,
 			);
+			$data[] = self::INVISIBLE_ERROR . 'This will break WebAuthn authentication.';
 		}
 
 		$config_url_imgage = config('filesystems.disks.images.url');
 		if ($config_url_imgage === '') {
 			$data[] = 'Error: LYCHEE_UPLOADS_URL is set and empty. This will prevent images to be displayed. Remove the line from your .env';
 		}
 
-		if (($config_url . '/uploads/') === $config_url_imgage && !$this->checkUrlMatchCurrentHost()) {
+		if (($config_url . $dir_url . '/uploads/') === $config_url_imgage && !$this->checkUrlMatchCurrentHost()) {
 			$data[] = sprintf(
-				'Error: APP_URL (%s) does not match the current url (%s). This will prevent images to be properly displayed.',
+				'Error: APP_URL (%s) does not match the current url (%s).',
 				$censored_app_url,
-				$censored_current);
+				$censored_current
+			);
+			$data[] = self::INVISIBLE_ERROR . 'This will prevent images to be properly displayed.';
```

Comment:
```suggestion
			$data[] = self::INVISIBLE_ERROR . 'This will prevent images from being properly displayed.';
```

I feel like I've fixed that already? Maybe it was in another PR :wink:.